openmeter/app/entity/app.go

```diff
@@ -0,0 +1,94 @@
+package appentity
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	appentitybase "github.com/openmeterio/openmeter/openmeter/app/entity/base"
+	customerentity "github.com/openmeterio/openmeter/openmeter/customer/entity"
+	"github.com/openmeterio/openmeter/pkg/pagination"
+)
+
+// App represents an installed app
+type App interface {
+	GetAppBase() appentitybase.AppBase
+	GetID() appentitybase.AppID
+	GetType() appentitybase.AppType
+	GetName() string
+	GetStatus() appentitybase.AppStatus
+	Get() appentitybase.MarketplaceListing
+
+	// ValidateCapabilities validates if the app can run for the given capabilities
+	ValidateCapabilities(capabilities []appentitybase.CapabilityType) error
+
+	// // ValidateCustomer validates if the app can run for the given customer
+	ValidateCustomer(ctx context.Context, customer *customerentity.Customer, capabilities []appentitybase.CapabilityType) error
```

Comment:
This should not be part of this interface; this is a generic app, not an app capable of customer sync.